src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java

```diff
@@ -142,10 +142,11 @@ default void damageRotor(int damageAmount) {
     default boolean isFrontFaceFree() {
         var facing = self().getFrontFacing();
         boolean permuteXZ = facing.getAxis() == Direction.Axis.Z;
+        boolean permuteY = facing.getAxis() == Direction.Axis.Y;
         var centerPos = self().getPos().relative(facing);
         for (int x = -1; x < 2; x++) {
             for (int y = -1; y < 2; y++) {
-                var blockPos = centerPos.offset(permuteXZ ? x : 0, y, permuteXZ ? 0 : x);
+                var blockPos = centerPos.offset(permuteXZ ? x : permuteY ? y : 0, permuteY ? 0 : y, permuteXZ ? 0 : x);
```

Comment:
shouldn't the Y axis check also be done for the Z offset?

Reply:
it plugs the value of x to the z offset value if you're facing towards the y-axis, so it does check the z offset

Reply:
not as far as I can see, the `permuteY` ternary is only done for the x and y values, not for Z
you never set X in the ternary

Reply:
yeah, but when you're facing the y-axis you're also not facing the z-axis so x is plugged in

Reply:
hold up does this work with extended facing at all rn

Reply:
nvm you're right

Reply:
does it matter?
the new version works just as well and there shouldn't be a reason to change this logic ever

Reply:
IMO it's cleaner